### DISCUSSION TITLE
Is it possible to configure SSL in arena.config.ts?

### DISCUSSION THREAD

**Discussion by cort1c** [colyseus/colyseus#728] [Q&A] [answered]
If not, what's the recommended way of doing it? It seems like I can create a *new* secure server in initializeTransport but I don't have a reference to the existing app so I end up having to create a new one that ignores everything configured in initializeExpress. That seems a little clumsy.

By the way, I know SSL is usually terminated at a web server like nginx but I may have some unique requirements that could require Colyseus/Express to do it.

**Comment by endel** [comment, marked as answer]
Hi @cort1c, that's not currently possible if you're using the `@colyseus/tools` package, as it internally uses `http.createServer` for the `express` layer: https://github.com/colyseus/colyseus/blob/49b010b339b03d848eab17699993bf9ee314b8e4/packages/tools/src/index.ts#L159-L160

To use terminate SSL at the Node.js layer you'd need to use `https.createServer()` and provide the certificates. Right now you can only do this by not using the `@colyseus/tools` package and initializing the Colyseus `Server` manually.

**Reply by cort1c** [reply]
That's what I suspected but thanks for confirming.
